src/components/MapView/DateSelector/index.tsx

```diff
@@ -43,38 +51,148 @@ function DateSelector({ availableDates = [], classes }: DateSelectorProps) {
+    ) {
+      setTimelinePosition({ x, y: 0 });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pointerPosition]);
+
+  // Create timeline range and set pointer position
+  useEffect(() => {
+    const range = Array.from(
+      moment()
+        .range(
+          moment(stateStartDate).startOf('year'),
+          moment(stateStartDate).endOf('year'),
+        )
+        .by('days'),
+    ).map(e => {
+      return {
+        value: e.valueOf(),
+        label: e.format('DD MMM YYYY'),
+        month: e.format('MMM YYYY'),
+        isFirstday: e.date() === e.startOf('month').date(),
+      };
+    });
+    setDateRange(range);
+    const dateIndex = findIndex(range, e => {
+      return e.label === moment(stateStartDate).format('DD MMM YYYY');
+    });
+    setPointerPosition({
+      x: dateIndex * 10,
+      y: 0,
+    });
     setSelectedDate(moment(stateStartDate));
   }, [stateStartDate]);
-  function updateMonth(month: number) {
-    const { startDate, endDate } = getMonthStartAndEnd(month, selectedYear);
-    dispatch(updateDateRange({ startDate, endDate }));
-  }
-  function incrementYear() {
-    setSelectedDate(selectedDate.clone().year(selectedYear + 1));
-  }
-  function decrementYear() {
-    setSelectedDate(selectedDate.clone().year(selectedYear - 1));
-  }
+
   function updateStartDate(date: Date) {
     const time = date.getTime();
     dispatch(updateDateRange({ startDate: time, endDate: time }));
   }
 
+  function setDatePosition(date: number, increment: number) {
+    const dates = availableDates.map(d => {
+      return d + USER_DATE_OFFSET;
+    });
+    const selectedIndex = findDateIndex(dates, date as number);
+    if (dates[selectedIndex + increment]) {
+      updateStartDate(new Date(dates[selectedIndex + increment]));
+    }
+  }
+
+  // move pointer to closest date when change map layer
+  if (!isEqual(dateRef.current, availableDates)) {
+    setDatePosition(stateStartDate as number, 0);
+    dateRef.current = availableDates;
+  }
+
+  function incrementDate() {
+    setDatePosition(stateStartDate as number, 1);
+  }
+
+  function decrementDate() {
+    setDatePosition(stateStartDate as number, -1);
+  }
+
+  // Click on available date to move the pointer
+  const clickDate = (index: number) => {
+    const dates = availableDates.map(date => {
+      return date + USER_DATE_OFFSET;
+    });
+    const selectedIndex = findDateIndex(dates, dateRange[index].value);
+    if (selectedIndex >= 0 && dates[selectedIndex] !== stateStartDate) {
+      setPointerPosition({ x: index * 10, y: 0 });
+      updateStartDate(new Date(dates[selectedIndex]));
+    }
+  };
+
+  // Set timeline position after being dragged
+  const onTimelineStop = (e: any, position: { x: number; y: number }) => {
+    setTimelinePosition(position);
+  };
+
+  // Set pointer position after being dragged
+  const onPointerStop = (e: any, position: { x: number; y: number }) => {
```

Comment:
You have a type definition for `{x: number, y: number}`. Moving it to a type variable and using it in your `useState` definitions and here also would clean things up.